### PR TITLE
Fix build on rust-nightly: bump dependencies where appropriate.

### PR DESCRIPTION
**Pull request by hfiguiere**
Tested with rustc 1.8.0-dev (c8fc4817d 2016-02-22)

@fabricedesre r?

(I had to bump clippy too)

### CODE REVIEW

**Comment by hfiguiere** [comment]
Seems that it fails with rustc from Feb 12... _sigh_

**Comment by fabricedesre** [comment]
Thanks for looking at that.
When testing locally with a fresh rustc (0de3cace0 2016-02-23), bumping clippy to 0.0.44 let us build on x86, but cross compiling still fails because of errors in the nix crate (actually a single error u8 vs i8 that repeats in a bunch of places).
So we need the nix crate fixed first.

**Comment by hfiguiere** [comment]
urgh. Ok, I can look at that.

**Comment by hfiguiere** [comment]
(I don't have a cross compiler since I don't have anything to run the output on. Will set one up)

**Comment by hfiguiere** [comment]
the problem is that ws pulls mio from crates that pull nix 0.4.2. while we pull mio from git that depends on nix-0.5.0 (git)

**Comment by hfiguiere** [comment]
And nix 0.5.0 builds fine....

**Comment by fabricedesre** [comment]
yes, this is why I used the git version of mio. I'm doing a fork of ws-rs to get us back on track.

**Comment by hfiguiere** [comment]
I started looking at ws and with jthe mio API change.... it is not that trivial. I'll let you continue with that instead. And then we can see about landing this change.

**Comment by hfiguiere** [comment]
I have ws working in my master branch https://github.com/hfiguiere/ws-rs

**Comment by fabricedesre** [comment]
Ok cool. Can you do a PR to point foxbox/Cargo.toml to your fork?

**Comment by hfiguiere** [comment]
@fabricedesre this one does (third commit). Or do you want something separate? I can squash things.

File: Cargo.toml [hunk 1]
```diff
@@ -22,10 +22,10 @@ router = "0.1.0"
 rustc-serialize = "0.3"
 serde = "0.6.13"
 serde_json = "0.6.0"
-serde_macros = "0.6.13"
+serde_macros = "0.6.14"
 staticfile = "0.1.0"
 uuid = "0.1.18"
-ws = "0.4.5"
+ws = { git = "https://github.com/hfiguiere/ws-rs.git" }
```
Comment (fabricedesre) [inline Cargo.toml]: We can switch back to upstream https://github.com/housleyjk/ws-rs.git since they merged your fixes.

**Comment by hfiguiere** [comment]
squahed and everything

**Pull request merged**